From: Iyappan Subramanian <isubramanian@apm.com>
Date: Thu, 18 May 2017 15:13:43 -0700
Subject: phy: Add helper function to check phy interface mode
Patch-mainline: v4.13-rc1
Git-commit: 32d0f7830d9be5b1652a718e050d808b4908155f
References: bsc#1033621

Added helper function that checks phy_mode is RGMII (all variants)
'bool phy_interface_mode_is_rgmii(phy_interface_t mode)'

Changed the following function, to use the above.
'bool phy_interface_is_rgmii(struct phy_device *phydev)'

[js] 4.4.112 already added the helper

Signed-off-by: Iyappan Subramanian <isubramanian@apm.com>
Suggested-by: Florian Fainelli <f.fainelli@gmail.com>
Suggested-by: Andrew Lunn <andrew@lunn.ch>
Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/linux/phy.h |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -699,8 +699,7 @@ static inline bool phy_interface_mode_is
  */
 static inline bool phy_interface_is_rgmii(struct phy_device *phydev)
 {
-	return phydev->interface >= PHY_INTERFACE_MODE_RGMII &&
-		phydev->interface <= PHY_INTERFACE_MODE_RGMII_TXID;
+	return phy_interface_mode_is_rgmii(phydev->interface);
 };
 
 /*
